version: "3.9"

services:
  postgres-db:
    container_name: postgres-analytics
    image: postgres:alpine3.19
    # build: .
    restart: always
    ports:
      - 5432:5432
    environment:
      POSTGRES_DB: northwind
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      PGDATA: /var/lib/postgresql/data
      TZ: America/Sao_Paulo
    volumes:
      - ./postgres-data:/var/lib/postgresql/data
      - ./postgresql_bin:/usr/lib/postgresql
      - ./northwind.sql:/docker-entrypoint-initdb.d/northwind.sql
      - ./files:/files
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 3s
      timeout: 5s
      retries: 10
    networks:
      - datanetwork

  adminer:
    container_name: adminer
    image: adminer:latest
    restart: always
    depends_on: 
      - postgres-db
    ports:
      - 8080:8080
    networks:
      - datanetwork

volumes:
  postgres-data:
    driver: local
  postgresql_bin:
    driver: local

networks:
  datanetwork:
    driver: bridge

